src/Interpreters/executeQuery.cpp

```diff
@@ -93,9 +93,25 @@ namespace
 /// broadcast the table definitions to notify all CatalogService
 void broadcastCatalogIfNecessary(const ASTPtr & ast, ContextPtr & context)
 {
-    if (!context->isDistributed() || context->isDistributedDDLOperation())
+    if (context->isDistributedDDLOperation())
```

Comment:
For all distributed DDL, we shall return directly ? in which case, we will need broadcast the catalogs ? What's the problem with previous logic ?

Reply:
1. Only in the first distributed DDL, we will return directly.  the second DDL will carry the distributed_ddl flag
2. When the second ddl  and the table  is DistributedMergeTree engine, and any time it is not DistributedMergeTree, it needs to be broadcast
3. The previous logic will cause catalog cannot be synchronized after non-DistributedMergeTree table been deleted through rest api, because the table is considered the first ddl

Reply:
if we are not in distributed env (no Kafka is deployed), we shall return directly ? `context->isDistributed()` is generally for this unless is temporary disabled. 

We actually don't quite care if the storage engine is DistributedMergeTree or plain MergeTree.

So for plain MergeTree, when it gets dropped via REST API (our REST API ?), what the process will be ?

Our rest api accepts the request, since it is MergeTree, do we send this request to dwal and then DDLService picks it up and then send the drop request to the target box as DistributedMergeTree does ?

Reply:
1. First of all, in the Interpreter processing layer, if it is 1.isDistributed 2.isDistributedDDLOperation 3.engine == "DistributedMergeTree"， when the three conditions are met at the same time, it will generate the first stage of DDL with Fakfa, and put the DDL information in Kafka for persistence Transform
2. The second stage is that ddlservice consumes data from the Kafka ddl_service topic and performs distributed processing. The sent request carries distributed_ddl = true to inform each node that this processing is the second stage

3. The current ddl processing process is that all ddl operations through REST API, if there is no parameter distributed_ddl = true in the request, it will be considered that the first stage of the interaction between Interpreter and kafka will be context->setDistributedDDLOperation (true), to avoid Doboardcast is unnecessary this time.

4. Drop mergree table Whether you are in distributed env or not, we all need to do a doboardcast to modify the catalog at this time.
5. When using REST API, because there is no distributed_ddl = true parameter, context->setDistributedDDLOperation(true) will be set, so that you can never enter the doboardcast

So here we need to separate DistributedMergeTree from plain MergeTree, and it doesn’t matter whether it’s distributed or not. 
Simply put, this bug is caused by the fact that the distributed_ddl = true parameter is not carried when operating the plain MergeTree through REST API.

Reply:
If it `isDistributed` is true, can we do the exact same CRUD process as DistributedMergeTree for MergeTree ?

Reply:
No we can't，We only support `DistributedMergeTree` table engine  in Interpreter  for now. We can leave a fixme here and wait until later to xact same CRUD process

Reply:
If we can't, we don't need fix this for now since in the future we will need revert this again ?

Reply:
No, we need fix this for now,  we don’t need evert this again later
1. First of all, we need to confirm that we only supports DistributedMergeTree table in REST API and DDL service
2. Only the DistributedMergeTree table will interact with kafka，so this place to judge whether it belongs to the second stage and engine is DistributedMergeTree
3.  plain MergeTree table also need to be board when doing ddl operations

Reply:
This may not be true. We probably need support MergeTree creation via DDL service